geocode_array/io_utils.py

```diff
@@ -127,8 +127,8 @@ def output_data(data, output_filename):
         data_writer_func(data_file, data)
 
 
-def get_api_key(key_filename):
+def get_api_key(key_filename, api_key):
```

Comment:
I think this it the best solution for handling per service keys.

Please update [the driver script](https://github.com/opendatadurban/geocode-array/blob/master/bin/geocode-array#L42) to do this correctly for Google (and use the Bing geocoder if it can). I don't think you should use the What3Words one by default, since their address format is pretty different from the others.

Reply:
Yeah, w3w will cause breaks since it doesn't take a normal address, only a three words address.

Have added to the driver script